crates/bitwarden/src/client/kdf.rs

```diff
@@ -22,6 +22,42 @@ pub enum Kdf {
     },
 }
 
+impl From<Kdf> for bitwarden_crypto::Kdf {
+    fn from(kdf: Kdf) -> Self {
+        match kdf {
+            Kdf::PBKDF2 { iterations } => bitwarden_crypto::Kdf::PBKDF2 { iterations },
+            Kdf::Argon2id {
+                iterations,
+                memory,
+                parallelism,
+            } => bitwarden_crypto::Kdf::Argon2id {
+                iterations,
+                memory,
+                parallelism,
+            },
+        }
+    }
+}
+
+impl From<&Kdf> for bitwarden_crypto::Kdf {
+    fn from(kdf: &Kdf) -> Self {
+        match kdf {
+            Kdf::PBKDF2 { iterations } => bitwarden_crypto::Kdf::PBKDF2 {
+                iterations: *iterations,
+            },
+            Kdf::Argon2id {
+                iterations,
+                memory,
+                parallelism,
+            } => bitwarden_crypto::Kdf::Argon2id {
+                iterations: *iterations,
+                memory: *memory,
+                parallelism: *parallelism,
+            },
+        }
+    }
+}
+
```

Comment:
This is pretty ugly, I think we may want to just not have a `TryFrom` for `PreloginResponseModel`, and instead use a normal function. That would allow the crypto crate to own the Kdf completely and we won't need to map between them.

Reply:
Is this because of the orphan rule? Yeah I think it makes sense to have our own function to do the conversion, we already follow that pattern with other types that have a `process_response` function.

Reply:
We own neither of the structs which means we can't implement traits on them.